madlibs.js

```diff
@@ -22,21 +22,151 @@
  *  { word: "store", pos: "noun" }
  *  { word: "," }
  *  ....
- * 
+ *
  * There are multiple ways to do this, but you may want to use regular expressions.
  * Please go through this lesson: https://www.freecodecamp.org/learn/javascript-algorithms-and-data-structures/regular-expressions/
  */
+
 function parseStory(rawStory) {
-  // Your code here.
-  return {}; // This line is currently wrong :)
+  const dot = '.';
+  const comma = ',';
+  const noun = '[n]';
+  const verb = '[v]';
+  const adjective = '[a]';
+  rawStory = rawStory.replace(dot, ' .');
+  rawStory = rawStory.replace(comma, ' , ');
+  const storyArray = rawStory.split(' ');
+  const objectArray = [];
+  storyArray.forEach((word) => {
+    const length = word.length - 3;
+    if (word.includes(noun)) {
```

Comment:
You haven't used regex to make this work, but still I love the ingenuity of making it work using your own way.